src/java/org/apache/cassandra/index/sai/disk/v1/postings/MergePostingList.java

```diff
@@ -36,15 +38,17 @@
 @NotThreadSafe
 public class MergePostingList implements PostingList
 {
-    final PriorityQueue<PeekablePostingList> postingLists;
-    final List<PeekablePostingList> temp;
+    private static final Comparator<PeekablePostingList> COMPARATOR = Comparator.comparingLong(PostingList.PeekablePostingList::peek);
+
+    final Collection<PeekablePostingList> postingLists;
+    // (Intersection code just calls advance(long), so don't create this until we need it)
```

Comment:
While this is true for the intersection code, how does it impact the union code that can be calling `nextPosting` and `advance`?